server/src/main/java/org/apache/gravitino/server/web/rest/RoleOperations.java

```diff
@@ -118,12 +120,29 @@ public Response getRole(@PathParam("metalake") String metalake, @PathParam("role
   @ResponseMetered(name = "create-role", absolute = true)
   public Response createRole(@PathParam("metalake") String metalake, RoleCreateRequest request) {
     try {
+
       return Utils.doAs(
           httpRequest,
           () -> {
+            Set<MetadataObject> metadataObjects = Sets.newHashSet();
             for (SecurableObjectDTO object : request.getSecurableObjects()) {
-              checkSecurableObject(metalake, object);
+              MetadataObject metadataObject =
+                  MetadataObjects.parse(object.getFullName(), object.type());
+              if (metadataObjects.contains(metadataObject)) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        "Don't support duplicated metadata objects %s type %s",
```

Comment:
Some places you're using securable objects, some places you're using metadata objects, I think you should unify them all.